### DISCUSSION TITLE
"common-external.yaml" throws 'namespace "rook-ceph-external" already exists' error

### DISCUSSION THREAD

**Discussion by roadSurfer** [rook/rook#13831] [Q&A]
I am attempting to connect to an external Rook cluster using the documented processes  and am receving an error:
* [Export Rook clustrer](https://www.rook.io/docs/rook/latest/CRDs/Cluster/external-cluster/#exporting-rook-to-another-cluster)
* [Manifest installation](https://www.rook.io/docs/rook/latest/CRDs/Cluster/external-cluster/#manifest-installation)

I paste in the values from `create-external-cluster-resources.py`, and run `. import-external-cluster.sh`

I deploy Rook: `kubectl create -f crds.yaml -f common.yaml -f operator.yaml`

And then check it has worked: `kubectl get namespaces`
```
NAME                 STATUS   AGE
default              Active   46h
kube-node-lease      Active   46h
kube-public          Active   46h
kube-system          Active   46h
rook-ceph            Active   11s
rook-ceph-external   Active   6m43s
```
As you can see, `rook-ceph-external` already exists.

Finally I attempt the connection: `kubectl create -f common-external.yaml -f cluster-external.yaml`
```
serviceaccount/rook-ceph-cmd-reporter created
role.rbac.authorization.k8s.io/rook-ceph-cmd-reporter created
cephcluster.ceph.rook.io/rook-ceph-external created
Error from server (AlreadyExists): error when creating "common-external.yaml": namespaces "rook-ceph-external" already exists
```

The cluster is shown: `kubectl -n rook-ceph-external get CephCluster`
```
NAME                 ...   AGE     PHASE       MESSAGE                          HEALTH      EXTERNAL ...
rook-ceph-external   ...   6m44s   Connected   Cluster connected successfully   HEALTH_OK   true ...
```
And: `kubectl -n rook-ceph-external get sc`
```
NAME       PROVISIONER                  RECLAIMPOLICY   VOLUMEBINDINGMODE   ALLOWVOLUMEEXPANSION   AGE
ceph-rbd   rook-ceph.rbd.csi.ceph.com   Delete          Immediate           true                   18m
```

I only did RBD on this occasion and I whilst can create/delete claims, I am just _slightly_ concerned that there is going to be an unpleasant surprise in my future.

Is the error a genuine problem, or just a spurious message?

**Comment by parth-gr** [comment]
Its just a warning that namespace exists and you are trying to re-create, it wont harm anything,

A quick fix for our code would be that remove the namespace creation from the `cluster-external.yaml`, @travisn do you agree?

**Reply by roadSurfer** [reply]
Just realised that if there is already a local Rook deployment, then `cluster-external.yaml` would need to create the new namespace. Although (being very new) I am not sure how likely that scenario is.

**Reply by parth-gr** [reply]
the new namespace is already created by the. import-external-cluster.sh, what I mean
